### PR TITLE
Made `RazorCodeGenerationOptionsBuilder.DesignTime` getter only

### PR DESCRIPTION
**Pull request by ajaybhargavb**
- Made RazorCodeGenerationOptions consistent with RazorParserOptions

#1510 

@NTaylorMullen @rynowak

### CODE REVIEW

File: src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIntermediateNodeLoweringPhase.cs [hunk 1]
```diff
@@ -13,11 +13,12 @@ namespace Microsoft.AspNetCore.Razor.Language
 {
     internal class DefaultRazorIntermediateNodeLoweringPhase : RazorEnginePhaseBase, IRazorIntermediateNodeLoweringPhase
     {
-        private IRazorCodeGenerationOptionsFeature[] _optionsCallbacks;
+        private IRazorCodeGenerationOptionsFeature _optionsFeature;
+
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIntermediateNodeLoweringPhase.cs]: extra line.

File: test/Microsoft.AspNetCore.Razor.Language.Test/DefaultRazorIntermediateNodeLoweringPhaseIntegrationTest.cs [hunk 2]
```diff
@@ -431,18 +433,35 @@ public void Lower_WithImports_IgnoresBlockDirective()
...
         {
             tagHelpers = tagHelpers ?? new TagHelperDescriptor[0];
 
-            var engine = RazorEngine.Create(b =>
+            RazorEngine engine = null;
+            if (designTime)
             {
-                builder?.Invoke(b);
+                engine = RazorEngine.CreateDesignTime(b =>
+                {
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Razor.Language.Test/DefaultRazorIntermediateNodeLoweringPhaseIntegrationTest.cs]: Lift this into a private function: `ConfigureEngine` and re-use.

File: test/Microsoft.AspNetCore.Razor.Test.Common/Language/CodeGeneration/TestCodeRenderingContext.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@ public static class TestCodeRenderingContext
         {
             var codeWriter = new CodeWriter();
             var documentNode = new DocumentIntermediateNode();
-            var options = RazorCodeGenerationOptions.CreateDesignTimeDefault();
+            var options = RazorCodeGenerationOptions.CreateDesignTime(c => { });
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Razor.Test.Common/Language/CodeGeneration/TestCodeRenderingContext.cs]: Ah, this is why you wanted a `CreateDesignTimeDefault`. It's a little ugly for tests but I think it's worth the lesser public API (people other than us will never need to do this).

As a side note, for lambdas where you don't use the variable it's common practice to use the `_` character

Comment (rynowak) [inline test/Microsoft.AspNetCore.Razor.Test.Common/Language/CodeGeneration/TestCodeRenderingContext.cs]: > As a side note, for lambdas where you don't use the variable it's common practice to use the _ character

This is absolutely not the case in C# and has never been part of our teams' guidelines. `_` is now a reserved word in some contexts, we shouldn't sprinkle it all over

File: src/Microsoft.AspNetCore.Razor.Language/RazorCodeGenerationOptions.cs [hunk 4]
```diff
@@ -1,23 +1,43 @@
...
+            }
+
+            var builder = new DefaultRazorCodeGenerationOptionsBuilder(designTime: false);
+            configure(builder);
+            var options = builder.Build();
+
+            return options;
         }
 
-        public static RazorCodeGenerationOptions CreateDesignTimeDefault()
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Language/RazorCodeGenerationOptions.cs]: 👎

File: test/Microsoft.AspNetCore.Razor.Language.Test/CodeGeneration/DefaultCodeTargetTest.cs [hunk 5]
```diff
@@ -31,7 +31,7 @@ public void Constructor_CreatesDefensiveCopy()
         public void CreateWriter_DesignTime_CreatesDesignTimeNodeWriter()
         {
             // Arrange
-            var options = RazorCodeGenerationOptions.CreateDesignTimeDefault();
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Razor.Language.Test/CodeGeneration/DefaultCodeTargetTest.cs]: Why take this away? If we're going to have `CreateDefault` we should have this

**Pull request merged**